diff --git a/frameworks/native/offlineaudioeffect/BUILD.gn b/frameworks/native/offlineaudioeffect/BUILD.gn
index 3c2460969..95247c0b1 100644
--- a/frameworks/native/offlineaudioeffect/BUILD.gn
+++ b/frameworks/native/offlineaudioeffect/BUILD.gn
@@ -19,6 +19,7 @@ config("offline_audio_effect_config") {
   include_dirs = [
     "include",
     "../../../interfaces/inner_api/native/offlineaudioeffect/include",
+    "../../../interfaces/inner_api/native/audioutils/include",
   ]
 
   cflags = [
@@ -86,6 +87,7 @@ if (!audio_framework_feature_offline_effect) {
       "include",
       "../../../interfaces/inner_api/native/audiocommon/include",
       "../../../services/audio_service/common/include",
+      "../../../interfaces/inner_api/native/audioutils/include",
     ]
 
     sources = [ "src/offline_audio_effect_server_chain.cpp" ]
diff --git a/frameworks/native/offlineaudioeffect/include/offline_audio_effect_chain_impl.h b/frameworks/native/offlineaudioeffect/include/offline_audio_effect_chain_impl.h
index 41ef94d29..09af8edab 100644
--- a/frameworks/native/offlineaudioeffect/include/offline_audio_effect_chain_impl.h
+++ b/frameworks/native/offlineaudioeffect/include/offline_audio_effect_chain_impl.h
@@ -40,6 +40,8 @@ public:
 
     void Release() override;
 
+    OfflineAudioEffectChainImpl(){};
+
     OfflineAudioEffectChainImpl(const std::string &chainName);
 
     ~OfflineAudioEffectChainImpl();
diff --git a/frameworks/native/offlineaudioeffect/include/offline_audio_effect_server_chain.h b/frameworks/native/offlineaudioeffect/include/offline_audio_effect_server_chain.h
index d16010918..bb1f1105d 100644
--- a/frameworks/native/offlineaudioeffect/include/offline_audio_effect_server_chain.h
+++ b/frameworks/native/offlineaudioeffect/include/offline_audio_effect_server_chain.h
@@ -21,6 +21,15 @@
 #include <string>
 #include <vector>
 #include <mutex>
+#include <algorithm>
+#include <chrono>
+#include <cinttypes>
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+#include <string>
+#include <unistd.h>
+#include "securec.h"
 
 #include "v1_0/effect_types.h"
 #include "v1_0/ieffect_control.h"
diff --git a/frameworks/native/offlineaudioeffect/test/unittest/BUILD.gn b/frameworks/native/offlineaudioeffect/test/unittest/BUILD.gn
index 438b85732..5c89e781f 100644
--- a/frameworks/native/offlineaudioeffect/test/unittest/BUILD.gn
+++ b/frameworks/native/offlineaudioeffect/test/unittest/BUILD.gn
@@ -28,6 +28,7 @@ if (!audio_framework_feature_offline_effect) {
       "../../../../../interfaces/inner_api/native/offlineaudioeffect/include",
       "../../../../../interfaces/inner_api/native/audiocommon/include",
       "../../../../../services/audio_service/client/include",
+      "../../../../../interfaces/inner_api/native/audioutils/include",
     ]
 
     cflags = [
@@ -43,6 +44,7 @@ if (!audio_framework_feature_offline_effect) {
       "../../:offline_audio_effect_service",
       "../../../../../services/audio_service:audio_client",
       "../../../audioutils:audio_utils",
+      "../../../../../services/audio_service:audio_common",
     ]
 
     external_deps = [
diff --git a/frameworks/native/offlineaudioeffect/test/unittest/offline_audio_effect_manager_unit_test.cpp b/frameworks/native/offlineaudioeffect/test/unittest/offline_audio_effect_manager_unit_test.cpp
index 1a514f988..da8912eef 100644
--- a/frameworks/native/offlineaudioeffect/test/unittest/offline_audio_effect_manager_unit_test.cpp
+++ b/frameworks/native/offlineaudioeffect/test/unittest/offline_audio_effect_manager_unit_test.cpp
@@ -20,7 +20,15 @@
 #include "audio_errors.h"
 #include "offline_audio_effect_manager.h"
 #include "offline_audio_effect_server_chain.h"
+#include "offline_audio_effect_chain_impl.h"
+#include "offline_stream_in_client.h"
 #include "audio_stream_info.h"
+#include "audio_shared_memory.h"
+#include "audio_log.h"
+#include "v1_0/effect_types.h"
+#include "v1_0/ieffect_control.h"
+#include "v1_0/ieffect_model.h"
+#include "audio_utils.h"
 
 using namespace testing::ext;
 using namespace testing;
@@ -35,6 +43,17 @@ namespace {
         AudioSamplingRate::SAMPLE_RATE_48000, AudioEncodingType::ENCODING_PCM,
         AudioSampleFormat::SAMPLE_S16LE, AudioChannel::STEREO, AudioChannelLayout::CH_LAYOUT_STEREO);
 }
+class IpcOfflineStreamMock : public IIpcOfflineStream {
+public:
+    MOCK_METHOD(int32_t, CreateOfflineEffectChain, (const std::string &chainName));
+    MOCK_METHOD(int32_t, ConfigureOfflineEffectChain, (const AudioStreamInfo& inInfo, const AudioStreamInfo& outInfo));
+    MOCK_METHOD(int32_t, SetParamOfflineEffectChain, (const std::vector<uint8_t>& param));
+    MOCK_METHOD(int32_t, PrepareOfflineEffectChain, (shared_ptr<AudioSharedMemory>& inBuffer, 
+        shared_ptr<AudioSharedMemory>& outBuffer));
+    MOCK_METHOD(int32_t, ProcessOfflineEffectChain, (uint32_t inSize, uint32_t outSize));
+    MOCK_METHOD(int32_t, ReleaseOfflineEffectChain, ());
+    MOCK_METHOD(sptr<IRemoteObject>, AsObject, ());
+};
 class OfflineAudioEffectManagerUnitTest : public testing::Test {
 public:
     static void SetUpTestCase(void);
@@ -71,6 +90,73 @@ void OfflineAudioEffectChainUnitTest::SetUpTestCase(void)
     }
 }
 
+unique_ptr<OfflineAudioEffectChain> OfflineAudioEffectManager::CreateOfflineAudioEffectChainMock(
+    const std::string &chainName)
+{
+    sptr<IpcOfflineStreamMock> mockProxy = new IpcOfflineStreamMock();
+    EXPECT_CALL(*mockProxy, PrepareOfflineEffectChain(_, _)).WillOnce(DoAll(
+        SetArgReferee<0>(AudioSharedMemory::CreateFormLocal(1, "testEffect")),
+        SetArgReferee<1>(AudioSharedMemory::CreateFormLocal(1, "testEffect")),
+        Return(0)
+    ));
+    unique_ptr<OfflineAudioEffectChainImpl> chain = std::make_unique<OfflineAudioEffectChainImpl>();
+    chain->chainName_ = chainName;
+    chain->offlineStreamInClient_ = make_shared<OfflineStreamInClient>(mockProxy);
+    int32_t ret = chain->CreateEffectChain();
+    CHECK_AND_RETURN_RET_LOG(ret == SUCCESS, nullptr, "create OfflineEffectChain failed, errcode is %{public}d", ret);
+    return chain;
+}
+  
+int32_t EffectProcess(struct IEffectControl *self, const struct AudioEffectBuffer* input,
+    struct AudioEffectBuffer* output) {
+    return SUCCESS;
+}
+
+int32_t SendCommand(struct IEffectControl *self, uint32_t cmdId, const int8_t* cmdData, uint32_t cmdDataLen,
+    int8_t* replyData, uint32_t* replyDataLen) {
+    return SUCCESS;
+}
+
+int32_t GetEffectDescriptor(struct IEffectControl *self, struct EffectControllerDescriptor* desc) {
+    return SUCCESS;
+}
+
+int32_t EffectReverse(struct IEffectControl *self, const struct AudioEffectBuffer* input,
+    struct AudioEffectBuffer* output) {
+    return SUCCESS;
+}
+
+int32_t GetVersion(struct IEffectControl *self, uint32_t* majorVer, uint32_t* minorVer) {
+    return SUCCESS;
+}
+
+int32_t IsSupplyEffectLibs(struct IEffectModel *self, bool* supply) {
+    return SUCCESS;
+}
+
+int32_t GetAllEffectDescriptors(struct IEffectModel *self, struct EffectControllerDescriptor* descs,
+    uint32_t* descsLen) {
+    return SUCCESS;
+}
+
+int32_t CreateEffectController(struct IEffectModel *self, const struct EffectInfo* info,
+    struct IEffectControl** contoller, struct ControllerId* id) {
+    return SUCCESS;
+}
+
+int32_t DestroyEffectController(struct IEffectModel *self, const struct ControllerId* id) {
+    return SUCCESS;
+}
+
+int32_t GetEffectDescriptor(struct IEffectModel *self, const char* effectId,
+    struct EffectControllerDescriptor* desc) {
+    return SUCCESS;
+}
+
+int32_t GetVersion(struct IEffectModel *self, uint32_t* majorVer, uint32_t* minorVer) {
+    return SUCCESS;
+}
+
 void OfflineAudioEffectChainUnitTest::TearDownTestCase(void)
 {
     g_manager = nullptr;
@@ -78,7 +164,7 @@ void OfflineAudioEffectChainUnitTest::TearDownTestCase(void)
 
 void OfflineAudioEffectChainUnitTest::SetUp(void)
 {
-    chain_ = g_manager->CreateOfflineAudioEffectChain(g_normalName);
+    chain_ = g_manager->CreateOfflineAudioEffectChainMock(g_normalName);
 }
 
  void OfflineAudioEffectChainUnitTest::TearDown(void)
@@ -122,25 +208,9 @@ HWTEST(OfflineAudioEffectManagerUnitTest, OfflineAudioEffectManager_001, TestSiz
 HWTEST(OfflineAudioEffectManagerUnitTest, OfflineAudioEffectManager_002, TestSize.Level1)
 {
     auto manager = make_shared<OfflineAudioEffectManager>();
-    auto chain = manager->CreateOfflineAudioEffectChain(g_normalName);
-    if (g_normalName == "") {
-        EXPECT_EQ(nullptr, chain);
-    } else {
-        EXPECT_NE(nullptr, chain);
-    }
-}
-
-/**
- * @tc.name  : Test CreateOfflineAudioEffectChain API
- * @tc.type  : FUNC
- * @tc.number: OfflineAudioEffectManager_003
- * @tc.desc  : Test OfflineAudioEffectManager interface.
- */
-HWTEST(OfflineAudioEffectManagerUnitTest, OfflineAudioEffectManager_003, TestSize.Level1)
-{
-    auto manager = make_shared<OfflineAudioEffectManager>();
-    auto chain = manager->CreateOfflineAudioEffectChain(INVALID_EFFECT_NAME);
-    EXPECT_EQ(nullptr, chain);
+    auto chain = manager->CreateOfflineAudioEffectChainMock(g_normalName);
+    EXPECT_NE(nullptr, chain);
+    EXPECT_EQ(SUCCESS, chain->Prepare());
 }
 
 /**
@@ -154,6 +224,7 @@ HWTEST_F(OfflineAudioEffectChainUnitTest, OfflineAudioEffectChain_001, TestSize.
     if (chain_) {
         int32_t ret = chain_->Configure(NORMAL_STREAM_INFO, NORMAL_STREAM_INFO);
         EXPECT_EQ(SUCCESS, ret);
+        EXPECT_EQ(SUCCESS, chain_->Prepare());
     }
 }
 
@@ -213,7 +284,8 @@ HWTEST_F(OfflineAudioEffectChainUnitTest, OfflineAudioEffectChain_004, TestSize.
     if (chain_) {
         std::vector<uint8_t> param(0);
         int32_t ret = chain_->SetParam(param);
-        EXPECT_EQ(ERROR, ret);
+        EXPECT_EQ(SUCCESS, ret);
+        EXPECT_EQ(SUCCESS, chain_->Prepare());
     }
 }
 
@@ -221,6 +293,13 @@ HWTEST_F(OfflineAudioEffectServerChainUnitTest, Create_001, TestSize.Level1)
 {
     std::shared_ptr<OfflineAudioEffectServerChain>  serverChain =
         std::make_shared<OfflineAudioEffectServerChain>("test");
+    IEffectControl *MockControl = new IEffectControl();
+    MockControl->EffectProcess = EffectProcess;
+    MockControl->SendCommand = SendCommand;
+    MockControl->GetEffectDescriptor = GetEffectDescriptor;
+    MockControl->EffectReverse = EffectReverse;
+    MockControl->GetVersion = GetVersion;
+    serverChain->controller_=MockControl;
     int32_t ret = serverChain->Create();
     EXPECT_EQ(ret, ERROR);
 }
@@ -240,8 +319,18 @@ HWTEST_F(OfflineAudioEffectServerChainUnitTest, SetConfig_001, TestSize.Level1)
     outInfo.encoding = AudioEncodingType::ENCODING_PCM;
     outInfo.format = AudioSampleFormat::SAMPLE_S16LE;
     outInfo.channels = AudioChannel::MONO;
+
     int32_t ret = serverChain->SetConfig(inInfo, outInfo);
     EXPECT_EQ(ret, ERROR);
+    IEffectControl *MockControl = new IEffectControl();
+    MockControl->EffectProcess = EffectProcess;
+    MockControl->SendCommand = SendCommand;
+    MockControl->GetEffectDescriptor = GetEffectDescriptor;
+    MockControl->EffectReverse = EffectReverse;
+    MockControl->GetVersion = GetVersion;
+    serverChain->controller_=MockControl;
+    ret = serverChain->SetConfig(inInfo, outInfo);
+    EXPECT_EQ(ret, SUCCESS);
 }
 
 HWTEST_F(OfflineAudioEffectServerChainUnitTest, SetParam_001, TestSize.Level1)
@@ -249,8 +338,18 @@ HWTEST_F(OfflineAudioEffectServerChainUnitTest, SetParam_001, TestSize.Level1)
     std::shared_ptr<OfflineAudioEffectServerChain>  serverChain =
         std::make_shared<OfflineAudioEffectServerChain>("test");
     std::vector<uint8_t> param(0);
+
     int32_t ret = serverChain->SetParam(param);
     EXPECT_EQ(ret, ERROR);
+    IEffectControl *MockControl = new IEffectControl();
+    MockControl->EffectProcess = EffectProcess;
+    MockControl->SendCommand = SendCommand;
+    MockControl->GetEffectDescriptor = GetEffectDescriptor;
+    MockControl->EffectReverse = EffectReverse;
+    MockControl->GetVersion = GetVersion;
+    serverChain->controller_=MockControl;
+    ret = serverChain->SetParam(param);
+    EXPECT_EQ(ret, SUCCESS);
 }
 
 HWTEST_F(OfflineAudioEffectServerChainUnitTest, GetEffectBufferSize_001, TestSize.Level1)
@@ -290,6 +389,14 @@ HWTEST_F(OfflineAudioEffectServerChainUnitTest, Release_001, TestSize.Level1)
 {
     std::shared_ptr<OfflineAudioEffectServerChain>  serverChain =
         std::make_shared<OfflineAudioEffectServerChain>("test");
+
+    IEffectControl *MockControl = new IEffectControl();
+    MockControl->EffectProcess = EffectProcess;
+    MockControl->SendCommand = SendCommand;
+    MockControl->GetEffectDescriptor = GetEffectDescriptor;
+    MockControl->EffectReverse = EffectReverse;
+    MockControl->GetVersion = GetVersion;
+    serverChain->controller_=MockControl;
     int32_t ret = serverChain->Release();
     EXPECT_EQ(ret, ERROR);
 }
diff --git a/interfaces/inner_api/native/offlineaudioeffect/include/offline_audio_effect_manager.h b/interfaces/inner_api/native/offlineaudioeffect/include/offline_audio_effect_manager.h
index b0c11f132..a8dcb6c7d 100644
--- a/interfaces/inner_api/native/offlineaudioeffect/include/offline_audio_effect_manager.h
+++ b/interfaces/inner_api/native/offlineaudioeffect/include/offline_audio_effect_manager.h
@@ -100,6 +100,8 @@ public:
      * @since 15
      */
     std::unique_ptr<OfflineAudioEffectChain> CreateOfflineAudioEffectChain(const std::string &chainName);
+
+    std::unique_ptr<OfflineAudioEffectChain> CreateOfflineAudioEffectChainMock(const std::string &chainName);
 };
 } // namespace AudioStandard
 } // namespace OHOS
diff --git a/services/audio_service/client/include/offline_stream_in_client.h b/services/audio_service/client/include/offline_stream_in_client.h
index 8febcdf14..37a58c8f0 100644
--- a/services/audio_service/client/include/offline_stream_in_client.h
+++ b/services/audio_service/client/include/offline_stream_in_client.h
@@ -29,6 +29,7 @@ public:
     static int32_t GetOfflineAudioEffectChains(std::vector<std::string> &effectChains);
     static std::shared_ptr<OfflineStreamInClient> Create();
 
+    OfflineStreamInClient() = default;
     OfflineStreamInClient(const sptr<IIpcOfflineStream> &ipcProxy);
     ~OfflineStreamInClient() = default;
 
